src/Git.hs

```diff
@@ -483,19 +525,24 @@ tryIntegrate message candidateRef candidateSha targetBranch testBranch alwaysAdd
       -- If not (i.e. the current master is the parent of the proposed commit
       -- and the approval type is not MergeAndDeploy) then we just take that
       -- commit as-is.
-      targetBranchSha <- checkout targetBranch
-      parentSha       <- getParent sha
-      newTip <- case parentSha of
-        Nothing -> pure $ Just sha
-        parent
-          | alwaysAddMergeCommit      -> merge sha message
-          | parent == targetBranchSha -> pure $ Just sha
-        _moreThanOneCommitBehind      -> merge sha message
-
-      -- If both the rebase, and the (potential) merge went well, push it to the
-      -- testing branch so CI will build it.
-      case newTip of
-        Just tipSha -> forcePush tipSha testBranch
-        Nothing     -> pure ()
-
-      pure newTip
+      checkOrphansResult <- checkOrphanFixups sha targetBranch
+      if checkOrphansResult 
+        then pure $ Left WrongFixups
+        else do
+          targetBranchSha <- checkout targetBranch
+          parentSha       <- getParent sha
+          newTip <- case parentSha of
+                      Nothing -> pure $ Just sha
+                      parent
+                        | alwaysAddMergeCommit      -> merge sha message
+                        | parent == targetBranchSha -> pure $ Just sha
+                      _moreThanOneCommitBehind      -> merge sha message
```

Comment:
Super-Nit: we try to prevent "excessive" blank space by only increasing indentation one step (of 2 spaces) at a time.

While this is also readable, having it consistent with the rest (and the previous version of this code) would be nice.